internal/jsnum/pseudobigint.go

```diff
@@ -25,3 +31,37 @@ func (value PseudoBigInt) Sign() int {
 	}
 	return 1
 }
+
+func ParseValidBigInt(text string) PseudoBigInt {
+	text, negative := strings.CutPrefix(text, "-")
+	return PseudoBigInt{
+		Negative:    negative,
+		Base10Value: ParsePseudoBigInt(text),
+	}
+}
+
+func ParsePseudoBigInt(stringValue string) string {
+	stringValue = strings.TrimSuffix(stringValue, "n")
+
+	var b1 byte
+	if len(stringValue) > 1 {
+		b1 = stringValue[1]
+	}
+
+	switch b1 {
+	case 'b', 'B', 'o', 'O', 'x', 'X':
+		// Not decimal.
+	default:
+		stringValue = strings.TrimLeft(stringValue, "0")
+		if stringValue == "" {
+			return "0"
+		}
+		return stringValue
+	}
+
+	bi, ok := new(big.Int).SetString(stringValue, 0)
+	if !ok {
+		panic(fmt.Sprintf("Failed to parse big int: %q", stringValue))
+	}
+	return bi.String() // !!!
```

Comment:
My super lazy way of not porting all of `ParsePseudoBigInt` right now. The Go bigint parser works fine for these.

Reply:
seems reasonable. wait, does this do base conversion from binary etc to decimal? Or does it just roundtrip whatever was written? The TODO comment in scanner.go implies the former.

Edit: And all the baseline changes.